nsot/api/views.py

```diff
@@ -571,6 +571,16 @@ class InterfaceViewSet(ResourceViewSet):
     queryset = models.Interface.objects.all()
     serializer_class = serializers.InterfaceSerializer
     filter_class = filters.InterfaceFilter
+    # Match on device_hostname:name or pk id
+    lookup_value_regex = ('[a-zA-Z0-9\-]+:[a-zA-Z][a-zA-Z0-9\/\-:.]*[0-9]'
```

Comment:
This looks like a great start for the regex but seeing as we currently don't have any naming constraints on `Interface.name` other than "non-empty", we should take this opportunity to enforce consistency between the two.

This means we should probably apply the same validation regex to interface names, so that you can't name an Interface in a way that would make it impossible to retrieve by natural key.

Lastly, we also should probably take a sampling of known interface names and make sure the regex pattern can support them. For example, interface names with spaces wouldn't be supported by this regex!